var/spack/repos/builtin/packages/elpa/package.py

```diff
@@ -69,25 +70,50 @@ def setup_environment(self, spack_env, run_env):
         spack_env.set('FC', spec['mpi'].mpifc)
         spack_env.set('CXX', spec['mpi'].mpicxx)
 
-        spack_env.append_flags('LDFLAGS', spec['lapack'].libs.search_flags)
-        spack_env.append_flags('LIBS', spec['lapack'].libs.link_flags)
-        spack_env.set('SCALAPACK_LDFLAGS', spec['scalapack'].libs.joined())
+        if spec.satisfies('%intel'):
+            spack_env.set('CPP', 'gcc -E')
```

Comment:
I'm wondering why not `icc -E` here, since we are using ICC in `spec`. This may cause problems when specifying ICC-only `cflags` such as `-xHost`.

Reply:
If I remember correctly, this was taken from the example `Makefile`s provided by cp2k.